augmentation/augmentation.hpp

```diff
@@ -0,0 +1,172 @@
+/**
+ * @file augmentation.hpp
+ * @author Kartik Dutt
+ * 
+ * Definition of Augmentation class for augmenting data.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/methods/ann/layer/bilinear_interpolation.hpp>
+#include <boost/regex.hpp>
+
+#ifndef MODELS_AUGMENTATION_HPP
+#define MODELS_AUGMENTATION_HPP
+
+/**
+ * Augmentation class used to perform augmentations / transform the data.
+ * For the list of supported augmentation, take a look at our wiki page.
+ *
+ * @code
+ * Augmentation<> augmentation({"horizontal-flip", "resize = (224, 224)"}, 0.2);
+ * augmentation.Transform(dataloader.TrainFeatures);
+ * @endcode
+ * 
+ * @tparam DatasetType Datatype on which augmentation will be done.
+ */
+template<typename DatasetType = arma::mat>
+class Augmentation
+{
+ public:
+  //! Create the augmentation class object.
+  Augmentation();
+
+  /**
+   * Constructor for augmentation class.
+   *
+   * @param augmentations List of strings containing one of the supported
+   *                     augmentations.
```

Comment:
Either this should be tabbed twice (4 spaces) or should align with the first line of the parameter description.

Reply:
Sure, will make the change here and elsewhere too. Thanks a lot.